src/commands/linkArticles.ts

```diff
@@ -83,7 +86,42 @@ async function insert(
 	if (templateString.includes("{description}") && !article.description)
 		new Notice("The article has no description.");
 
+	if (template.createArticleNote) {
+		if (!template.createArticleNoteUseCustomPath) {
+			template.createArticleNoteCustomPath = settings.createArticleNotePath;
+		}
+		const createdNote = await createNoteInFolder(app, noteTitle, insert, template);
+		if (createdNote != null) {
+			insert = `[[${createdNote}|${noteTitle}]]`;
+		}
+	}
+
 	const cursorPosition = editor.getCursor();
 	editor.replaceSelection(insert);
 	if (settings.placeCursorInfrontOfInsert) editor.setCursor(cursorPosition);
```

Comment:
If the note couldn't be created nothing should happen. Don't just insert the content that is meant to be in a separate file directly inside the current file.